### PR TITLE
Drop down to choose which fiat on-ramp 

### LINKED ISSUE #6955: Drop down to choose which fiat on-ramp
as we will have a few more providers as backup
- #6946 
- #6727

we will be looking to get drop down to select which will be primary and secondary

### LINKED ISSUE #7164: a bit shorter on-ramp dialog for Transak
> Loads, however it could be a bit shorter. 

image

_Originally posted by @yangwao in https://github.com/kodadot/nft-gallery/issues/7159#issuecomment-1706981192_

### PR DESCRIPTION
**Pull request by hassnian**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #6955
- [x] Closes #7164

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

CleanShot 2023-09-06 at 15 19 04@2x

CleanShot 2023-09-06 at 15 19 20@2x

### <samp>🤖 Generated by Copilot at 2300db2</samp>

The pull request refactors and improves the functionality for adding funds to the wallet. It creates a new component `OnRampModal` that displays a list of providers that support ramp integration, and uses the `useTransak` composable to handle the Transak widget. It also fixes some minor issues with the import path and the UI of the widget, and adds some translations and styles for the modal.

### <samp>🤖 Generated by Copilot at 2300db2</samp>

> _Sing, O Muse, of the skillful coder who refactored_
> _The wallet's ramp, and created a new component `OnRampModal`_
> _Which shows the users a list of providers, like Transak_
> _The swift and easy way to add funds to their account._

### CODE REVIEW

**Comment by exezbcz** [comment]
should not be a cursor pointer over disabled services 
image

otherwise works for me!

File: components/shared/OnRampModal.vue [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  })
+}
+</script>
+
+<style lang="scss" scoped>
+@import '@/styles/abstracts/variables';
+
+.small-text {
+  font-size: 0.75rem;
+}
```
Comment (roiLeo) [inline components/shared/OnRampModal.vue:144]: Bulma [typography helpers](https://bulma.io/documentation/helpers/typography-helpers/) might help

File: components/shared/OnRampModal.vue [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+.provider {
+  cursor: pointer;
```
Comment (roiLeo) [inline components/shared/OnRampModal.vue:147]: `is-clickable` from https://bulma.io/documentation/helpers/other-helpers/
or change `<div>` tag to `<button>`

File: components/shared/OnRampModal.vue [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  }
+
+  &__disabled {
+    cursor: default;
+    .provider-logo {
+      opacity: 30%;
+    }
+  }
+
+  .devider {
```
Comment (roiLeo) [inline components/shared/OnRampModal.vue:160]: can't we reuse something we already done? + typo on class

File: composables/useTransak.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import transakSDK from '@transak/transak-sdk'
 import { decodeAddress, encodeAddress } from '@polkadot/util-crypto'
-import { CHAINS } from '~~/libs/static/dist'
+import { CHAINS } from '@/libs/static/src/chains'
```
Comment (roiLeo) [inline composables/useTransak.ts:3]:
```suggestion
import { CHAINS } from '@kodadot1/static'
```

**Review by roiLeo — COMMENTED** [review]
Small stuff

**Review by roiLeo — APPROVED** [review]
✅ code lgtm

**Comment by yangwao** [comment]
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 4.23 USD/DOT ~ 11.82 $DOT
🧗 [13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh](https://polkadot.subscan.io/account/13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)
🔗 [0xf49fd8d3901e384a4b982e8218cd11111897cac1b36f8c223060787e8d757026](https://polkadot.subscan.io/extrinsic/0xf49fd8d3901e384a4b982e8218cd11111897cac1b36f8c223060787e8d757026)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**